data-access/src/app/application-services/roles/end-user-role/end-user-role.data.ts

```diff
@@ -13,28 +13,28 @@ export class EndUserRoleDataApiImpl extends CosmosDataSource<EndUserRoleData, Ap
 
   async getRoleById(id: string): Promise<EndUserRoleData> {
     const roles = await this.findByFields({ community: this.context.community?.id });
-    const roleToReturn = roles.find(role => role.id === id);
-    if (roleToReturn && this.applyPermissions(roleToReturn)) {
-      return roleToReturn;
-    }
+    return roles.find(role => role.id === id);
+    // if (roleToReturn && this.applyPermissions(roleToReturn)) {
+    //   return roleToReturn;
+    // }
   }
 
   async getRoles(): Promise<EndUserRoleData[]> {
-    const rolesToReturn = await this.findByFields({ community: this.context.community?.id });
-    rolesToReturn.filter(roleData => this.applyPermissions(roleData));
```

Comment:
We are moving away from permissions in the datasource functions and allowing them to be called anywhere in the backend. We are moving the permissions checks to the graphql query and mutation resolvers which call these underlying datasource methods to perform their action. In the case of roles, I previously added permission checks to the role subfield on Member graphql type in member.resolvers.ts to prevent unauthorized access to roles. So this code is safe to remove and intended

Reply:
Just confirmed roles are protected so members can only see their own role and admins can see roles on other members in their community.